python/ray/serve/controller.py

```diff
@@ -160,7 +160,7 @@ async def __init__(
 
         # Manage all applications' state
         self.application_state_manager = ApplicationStateManager(
-            self.deployment_state_manager
+            self.deployment_state_manager, self.endpoint_state, self.kv_store
```

Comment:
For my understanding, so the only reason to pass self.deployment_state_manager is that we have to keep the v1 api, right?

Reply:
Do you mean the reason why we don't construct the deployment state manager inside application state manager? If so, we would need to change the structure to be one deployment state manager per application. If we remove the v1 api, we can make that change if we want.